### PR TITLE
bootchart: Account CPU time spent in non-main threads of the process …

### PR DESCRIPTION
**Pull request by gmacario**
…(v4)

See https://github.com/systemd/systemd/issues/139
- Implement fixes suggested by @teg to -v2
- Implement fixes suggested by @zonque to -v3

### CODE REVIEW

File: src/bootchart/store.c [hunk 1]
```diff
@@ -409,6 +410,60 @@ int log_sample(DIR *proc,
...
+                 * See https://github.com/systemd/systemd/issues/139
+                 */
+
+                /* Browse directory "/proc/[pid]/task" to know the thread ids of process [pid] */
+                snprintf(filename, sizeof(filename), PID_FMT "/task", pid);
+                taskfd = openat(procfd, filename, O_RDONLY|O_DIRECTORY|O_CLOEXEC);
+                if (taskfd >= 0) {
+                        _cleanup_closedir_ DIR *taskdir = NULL;
+
+                        taskdir = fdopendir(taskfd);
```
Comment (zonque) [inline src/bootchart/store.c:424]:
```
if (!taskdir) {
    safe_close(taskfd);
    return -errno;
}
```

File: src/bootchart/store.c [hunk 2]
```diff
@@ -409,6 +410,60 @@ int log_sample(DIR *proc,
...
+                                r = safe_atolli(rt, &delta_rt);
+                                if (r < 0)
+                                    continue;
+                                r = safe_atolli(rt, &delta_wt);
+                                if (r < 0)
+                                    continue;
+                                ps->sample->runtime  += delta_rt;
+                                ps->sample->waittime += delta_wt;
+                        }
+                        taskfd = -1;
```
Comment (zonque) [inline src/bootchart/store.c:464]: This assignment is now unneeded.

**Comment by zonque** [comment]
Two more nits, that this is finally good to go in :)

**Comment by gmacario** [comment]
v5 of the Pull Request available in https://github.com/systemd/systemd/pull/366

**Pull request closed without merging**